lib/user.js

```diff
@@ -46,4 +46,25 @@ export default class User {
   requestPermanentDeletion(intercom_user_id, f) {
     return this.client.post('/user_delete_requests', { intercom_user_id }, f);
   }
+  requestPermanentDeletionByParams(params, f) {
+    if (params.id) {
+      return this.requestPermanentDeletion(params.id, f);
+    }
+
+    return this.find(params)
+      .then((res) => this.requestPermanentDeletion(res.body.id))
+      .then((res) => {
+        if (f) {
+          return f(null, res);
+        }
+
+        return res;
+      }).catch((err) => {
+        if (f) {
+          return f(err);
+        }
+
+        throw err;
+      });
```

Comment:
Rather than handling the success/error cases for callback and promises here, could we pass `f` into `requestPermanentDeletion` and let it handle that for us?

```node
return this.find(params).then((res) => this.requestPermanentDeletion(res.body.id, f))
```

Reply:
I don't think so as if `find` errors with `f` as a callback, `f` wouldn't get called with the error - so you need to have the `catch` block with the logic around the callback being given or not. 

I think you're right that the `then` block isn't needed though.

Reply:
Ah good point, I wasn't considering `find`. This looks good to me 👍